fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -641,6 +644,9 @@ struct CommitBatchContext {
 	std::set<Tag> writtenTags; // final set tags written to in the batch
 	std::set<Tag> writtenTagsPreResolution; // tags written to in the batch not including any changes from the resolver.
 
+	// Cipher keys to be used to encrypt mutations
+	std::unordered_map<EncryptCipherDomainId, Reference<BlobCipherKey>> cipherKeys;
```

Comment:
Any reason we are not using `BlobCipherDomainCacheMap`, it stores the same information

`using BlobCipherDomainCacheMap = std::unordered_map<EncryptCipherDomainId, Reference<BlobCipherKeyIdCache>>;`

Reply:
BlobCipherDomainCacheMap is a map of map, and cipherKeys maps domain id to cipher keys.

As for why not getting cached key directly from the BlobCipherDomainCacheMap singleton, we had discussion before where the BlobCipherDomainCacheMap needs to handle expiration and eviction in the future. Here in the commit proxy I need a reference to the cipher keys that's valid until I'm done using them.

Reply:
_BlobCipherDomainCacheMap_ allows one to query based on domainId and in your usecase that is what is needed I think. 
However, to your point about _caching locally to avoid rpc calls_, I mean it is a hit/miss, possible that your construction of local cache is the trigger to refresh cached value (due to expiration); given _BlobCipherKeyCache_ is a process local hashMap, I would suggest not to create another level of caching, instead use it directly. 

Further, KeyRefresh only leads to RPC call to EKP server, as EKP server refreshes interval is shorter than Cache TTL (8 mins versus 10 mins) to ensure that FDB processes KeyFetch can be served within FDB cluster processes for _inline write paths_ avoiding KMS round trip time.

Reply:
Let me explain my usage. On the high level CP process each commit batch in following steps (if you look at `commitBatch()` method in CommitProxyServer.actor.cpp there's inline comment explaining this):
1. Pre-resolution process (CPU bound)
2. Resolution (wait on network)
3. Post-resolution process (CPU bound)
4. Logging (wait on network)
5. Reply

For encryption, I'm nesting EKP request on phase 2, and the result is store in the local `cipherKeys` map. The map is used in phase 3 when the mutations are being iterator through. This way in phase 3 we avoid making async call to EKP, since the code for phase 3 is assumed to be mostly cpu computations. So here `cipherKeys` is the local cipher key result for the current commit batch. It is not like another layer of cache.